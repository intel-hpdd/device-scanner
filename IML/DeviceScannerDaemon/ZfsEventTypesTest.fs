module IML.DeviceScannerDaemon.ZFSEventTypesTest

open IML.DeviceScannerDaemon.TestFixtures
open ZFSEventTypes
open Fable.Import.Jest
open Matchers

let poolCreateMatch = function
  | ZedPool "create" x -> Some x
  | _ -> None

let poolDestroyMatch = function
  | ZedDestroy x -> Some x
  | _ -> None

let poolExportMatch = function
  | ZedExport x -> Some x
  | _ -> None

let poolImportMatch = function
  | ZedPool "import" x -> Some x
  | _ -> None

let datasetCreateMatch = function
  | ZedDataset "create" x -> Some x
  | _ -> None

let datasetDestroyMatch = function
  | ZedDataset "destroy" x -> Some x
  | _ -> None

test "Matching Events" <| fun () ->
  expect.assertions 6

  toMatchSnapshot (poolCreateMatch createZpool)

  toMatchSnapshot (poolDestroyMatch destroyZpool)

  toMatchSnapshot (poolExportMatch exportZpool)

  toMatchSnapshot (poolImportMatch importZpool)

  toMatchSnapshot (datasetCreateMatch createZdataset)

  toMatchSnapshot (datasetDestroyMatch destroyZdataset)
