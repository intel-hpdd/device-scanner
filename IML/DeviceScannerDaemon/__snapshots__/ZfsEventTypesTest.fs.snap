// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matching Events 1`] = `
ZfsPool {
  "DATASETS": FableMap {
    "comparer": Comparer {
      "Compare": [Function],
    },
    "tree": MapTree {
      "data": undefined,
      "tag": 0,
    },
  },
  "NAME": "testPool1",
  "PATH": "testPool1",
  "PROPERTIES": FableMap {
    "comparer": Comparer {
      "Compare": [Function],
    },
    "tree": MapTree {
      "data": undefined,
      "tag": 0,
    },
  },
  "STATE_STR": "ACTIVE",
  "UID": "0x2D28F440E514007F",
}
`;

exports[`Matching Events 2`] = `
ZfsPool {
  "DATASETS": FableMap {
    "comparer": Comparer {
      "Compare": [Function],
    },
    "tree": MapTree {
      "data": undefined,
      "tag": 0,
    },
  },
  "NAME": "testPool1",
  "PATH": "testPool1",
  "PROPERTIES": FableMap {
    "comparer": Comparer {
      "Compare": [Function],
    },
    "tree": MapTree {
      "data": undefined,
      "tag": 0,
    },
  },
  "STATE_STR": "DESTROYED",
  "UID": "0x2D28F440E514007F",
}
`;

exports[`Matching Events 3`] = `
ZfsPool {
  "DATASETS": FableMap {
    "comparer": Comparer {
      "Compare": [Function],
    },
    "tree": MapTree {
      "data": undefined,
      "tag": 0,
    },
  },
  "NAME": "testPool1",
  "PATH": "testPool1",
  "PROPERTIES": FableMap {
    "comparer": Comparer {
      "Compare": [Function],
    },
    "tree": MapTree {
      "data": undefined,
      "tag": 0,
    },
  },
  "STATE_STR": "EXPORTED",
  "UID": "0x2D28F440E514007F",
}
`;

exports[`Matching Events 4`] = `
ZfsPool {
  "DATASETS": FableMap {
    "comparer": Comparer {
      "Compare": [Function],
    },
    "tree": MapTree {
      "data": undefined,
      "tag": 0,
    },
  },
  "NAME": "testPool1",
  "PATH": "testPool1",
  "PROPERTIES": FableMap {
    "comparer": Comparer {
      "Compare": [Function],
    },
    "tree": MapTree {
      "data": undefined,
      "tag": 0,
    },
  },
  "STATE_STR": "ACTIVE",
  "UID": "0x2D28F440E514007F",
}
`;

exports[`Matching Events 5`] = `
ZfsDataset {
  "DATASET_NAME": "testPool1/home",
  "DATASET_UID": "11",
  "POOL_UID": "0x2D28F440E514007F",
  "PROPERTIES": FableMap {
    "comparer": Comparer {
      "Compare": [Function],
    },
    "tree": MapTree {
      "data": undefined,
      "tag": 0,
    },
  },
}
`;

exports[`Matching Events 6`] = `
ZfsDataset {
  "DATASET_NAME": "testPool1/home",
  "DATASET_UID": "11",
  "POOL_UID": "0x2D28F440E514007F",
  "PROPERTIES": FableMap {
    "comparer": Comparer {
      "Compare": [Function],
    },
    "tree": MapTree {
      "data": undefined,
      "tag": 0,
    },
  },
}
`;
